lib/app.js

```diff
@@ -28,29 +28,100 @@ app.get('/api/test', (req, res) => {
   });
 });
 
-app.get('/characters', async(req, res) => {
-  try {
-    const data = await client.query('SELECT * from characters');
+// app.get('/characters', async(req, res) => {
+//   try {
+//     const data = await client.query('SELECT * from characters');
     
-    res.json(data.rows);
-  } catch(e) {
+//     res.json(data.rows);
+//   } catch(e) {
     
+//     res.status(500).json({ error: e.message });
+//   }
+// });
+
+app.post('/characters', async(req, res)=>{
+  
+  try {
+    const data = await client.query(`
+    INSERT INTO characters(
+      id,
+      name,
+      bad,
+      species
+    ) VALUES ($1, $2, $3, $4) 
+    RETURNING *;`, [
+      req.body.id,
+      req.body.name,
+      req.body.bad,
+      req.body.species
+    ]);
+    res.json(data.rows[0]);
+
+  }catch(e){
     res.status(500).json({ error: e.message });
   }
 });
 
-app.get('/characters/:id', async(req, res) => {
-  const id = req.params.id;
+app.put('/characters/:id', async(req, res)=>{
   try {
-    const data = await client.query('SELECT * from characters WHERE id = $1', [id]);
-    
-    res.json(data.rows);
+    const data = await client.query(`
+      UPDATE characters
+      SET 
+        name=$2,
+        bad=$3,
+        species=$4
+      WHERE id = $1
+      RETURNING *;
+    `, [
+      req.params.id,
+      req.body.name,
+      req.body.bad,
+      req.body.species
+    ]);
+    res.json(data.rows[0]);
   } catch(e) {
-    
+    res.status(500).json({ error: e.message });
+  }
+});
+
+app.put('/characters/:id', async(req, res)=>{
```

Comment:
This should be a POST route instead of a PUT as the SQL query you're writing is for a POST. Remember the CRUD acronym: 

> C(reate) -> POST
> R(ead) -> GET
> U(pdate) -> PUT
> D(elete) -> DELETE

That acronym helped me really understand the different routes and I kept it handy as a reminder until I was able to fully understand them.